### DISCUSSION TITLE
Building Bun on Apple Silicon MacBook

### DISCUSSION THREAD

**Discussion by vjpr** [oven-sh/bun#4017] [General]
Does anyone build Bun on an M1 or similar MacBook?

First time my laptop crashed and rebooted. Next time it locked up my entire computer.

---

Seems changing `CPU_COUNT` in `Makefile` helps. Leaving it at the default 2 didn't overheat my laptop...

**Comment by Jarred-Sumner** [comment]
Most of the core team use MacBook Pro's on Apple Silicon

But you'll realistically need 16 GB of ram to compile Bun. It can work on 8 GB. But it comes close to the limit

**Comment by branaway** [comment]
why am I seeing:

warn: CPU lacks AVX support, strange crashes may occur. Reinstall Bun or use *-baseline build:
  https://github.com/oven-sh/bun/releases/download/bun-v1.2.17/bun-darwin-x64-baseline.zip

I'm running the latest 1.2.18 on my MacBook Pro m1

**Reply by RiskyMH** [reply]
@branaway not sure how you got the x64 version, but try the aarch64 build that's for arm/ apple silicon

https://github.com/oven-sh/bun/releases/download/bun-v1.2.18/bun-darwin-aarch64.zip

**Reply by branaway** [reply]
thanks for the information, but I have installed directly from the web using the curl command. I have also tried installing with brew, the same strange error.  I'll investigate further.

**Reply by branaway** [reply]
ha, I have found out that pnpm installs bun.exe if I have listed bun as a dev dependency.
